test/base/src/androidTest/kotlin/io/realm/kotlin/test/shared/MutableRealmIntTests.kt

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright 2022 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.test.shared
+
+import io.realm.kotlin.Realm
+import io.realm.kotlin.RealmConfiguration
+import io.realm.kotlin.entities.Sample
+import io.realm.kotlin.internal.RealmObjectReference
+import io.realm.kotlin.test.assertFailsWithMessage
+import io.realm.kotlin.test.platform.PlatformUtils
+import io.realm.kotlin.types.MutableRealmInt
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertNotEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNotSame
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+class MutableRealmIntTests {
+
+    private lateinit var tmpDir: String
+    private lateinit var realm: Realm
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = PlatformUtils.createTempDir()
+        val configuration = RealmConfiguration.Builder(setOf(Sample::class))
+            .directory(tmpDir)
+            .build()
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (this::realm.isInitialized && !realm.isClosed()) {
+            realm.close()
+        }
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    fun unmanaged_basic() {
+        val sample1 = Sample().apply { nullableMutableRealmIntField = null }
+        val sample2 = Sample().apply { nullableMutableRealmIntField = null }
+        basicTest(sample1, sample2)
+    }
+
+    @Test
+    fun unmanaged_equality() {
+        val sample1 = Sample()
+        val sample2 = Sample()
+        equalityTest(sample1, sample2)
+    }
+
+    @Test
+    fun unmanaged_nullability() {
+        nullabilityTest(Sample())
+    }
+
+    @Test
+    fun unmanaged_compareTo() {
```

Comment:
I'm missing specific tests for `increment`, `decrement`, especially corner cases like overflow/underflow, what happens if increment is given a negative number, and things like that.

Also specific tests for `set`